helper/fields/data.go

```diff
@@ -0,0 +1,166 @@
+package fields
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/mitchellh/mapstructure"
+)
+
+// FielData contains the raw data and the schema that the data should adhere to
+type FieldData struct {
+	Raw    map[string]interface{}
+	Schema map[string]*FieldSchema
+}
+
+// Cycle through raw data and validate conversions in the schema. Also check
```

Comment:
Start function comments with the name of the function. `Validate cycles through....`